plugins/inputs/snmp/snmp.go

```diff
@@ -535,8 +538,8 @@ func (t Table) Build(gs snmpConnection, walk bool) (*RTable, error) {
 		Time: time.Now(), //TODO record time at start
 		Rows: make([]RTableRow, 0, len(rows)),
 	}
-	for _, r := range rows {
-		if len(r.Tags) < tagCount {
+	for i, r := range rows {
+		if len(r.Tags)+emptyTags[i] < tagCount {
 			// don't add rows which are missing tags, as without tags you can't filter
 			continue
 		}
```

Comment:
Ah, I think I see what you're after now. I'm on the fence about whether we should treat missing tag differently from empty tag. In the end, both result in a data point getting inserted with no tag, and the original intent of the dropping was to prevent data from getting inserted and lumped into a single series (no ability to filter & separate them).
@sparrc What behavior would you prefer?
1. Drop data point if a tag is missing or empty *(current behavior)*
2. Drop data point if a tag is missing, but allow if empty, even though when it gets to influxdb, the tag will still be missing because empty tags are removed *(this PR)*
3. Allow data points with both missing and empty tags *(PR #2172)*

While I'm on the fence, I think I lean more towards # 3. I think I feel that if we want the behavior of # 2, it should be a generic telegraf option, similar to the `tagpass`/`tagdrop` options.